transport/src/main/java/io/scalecube/transport/Transport.java

```diff
@@ -284,55 +284,46 @@ public final void stop(@Nullable SettableFuture<Void> promise) {
 
   @Override
   public void send(@CheckForNull Address address, @CheckForNull Message message) {
-    send(address, message, null);
+    send(address, message, COMPLETED_PROMISE);
   }
 
   @Override
   public void send(@CheckForNull final Address address, @CheckForNull final Message message,
-      @Nullable final SettableFuture<Void> promise) {
+      @CheckForNull final CompletableFuture<Void> promise) {
     checkState(!stopped, "Transport is stopped");
     checkArgument(address != null);
     checkArgument(message != null);
+    checkArgument(promise != null);
     message.setSender(this.address);
 
     final ChannelFuture channelFuture = outgoingChannels.get(address);
     if (channelFuture.isSuccess()) {
       composeFutures(channelFuture.channel().writeAndFlush(message), promise);
     } else {
-      channelFuture.addListener(new ChannelFutureListener() {
-        @Override
-        public void operationComplete(ChannelFuture channelFuture) {
-          if (channelFuture.isSuccess()) {
-            composeFutures(channelFuture.channel().writeAndFlush(message), promise);
+      channelFuture.addListener((ChannelFuture chFuture) -> {
+          if (chFuture.isSuccess()) {
+            composeFutures(chFuture.channel().writeAndFlush(message), promise);
           } else {
-            if (promise != null) {
-              promise.setException(channelFuture.cause());
-            }
+            promise.completeExceptionally(chFuture.cause());
```

Comment:
As I understand from spec nothing should happen, this method provides effect only on uncompleted futures